tests/logging/contexts.py

```diff
@@ -0,0 +1,35 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+
+from __future__ import annotations
+
+import logging
+from contextlib import contextmanager
+from typing import Iterator
+
+from beamlime.constructors import Factory, ProviderGroup
+from beamlime.ready_factory import log_factory
+
+
+@contextmanager
+def local_logger_factory(*local_log_providers: ProviderGroup) -> Iterator[Factory]:
+    """
+    Keep a copy of logger names in logging.Logger.manager.loggerDict
+    and remove newly added loggers at the end of the context
+    within the sub context using ``log_factory``.
+
+    It will help a test not to interfere other tests.
+    """
+
+    original_logger_names = list(logging.Logger.manager.loggerDict.keys())
+    try:
+        with log_factory.local_factory(*local_log_providers) as local_log_factory:
+            yield local_log_factory
+    finally:
```

Comment:
It cleans up any loggers that were added to the `logging.Logger.manager` within this context.
It was easier than resetting handlers of each loggers. It also keeps original loggers that are unrelated to the tests as they were.